### PR TITLE
update CODEOWNERS

### PR DESCRIPTION
**Pull request by alexrashed**
## Motivation
We are more and more using the `CODEOWNERS` file in order to define code ownership in a machine readable way, which then can be used by different tools.
This can be GitHub by using it to auto-assigning PR reviewers, but it can also be used by tooling on our side in order to implement some kind of automation / notifications / ... around it.
This PR is another iteration on reducing the amount of uncovered files in this repo.
The enhancements were driven by the validation results created with [mszostok/codeowners-validator](https://github.com/mszostok/codeowners-validator).

## Changes
- Restructure the CODEOWNERS file a bit.
- Add a few more sections / directories / files.

/cc affected codeowners @HarshCasper @thrau @dfangl @dominikschubert @steffyP

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -5,6 +5,26 @@
...
 
+# README / Docs
+/doc/ @HarshCasper
+/README.md @HarshCasper
+/CODE_OF_CONDUCT.md @HarshCasper
+/CONTRIBUTING.md @HarshCasper
+/LICENSE.txt @HarshCasper @alexrashed
+
+# Docker
+/bin/docker-entrypoint.sh
```
Comment (thrau) [inline CODEOWNERS:16]: i'll take this one :-)

File: CODEOWNERS [hunk 2]
```diff
@@ -33,27 +53,27 @@
...
 
 # HTTP framework
 /localstack/http/ @thrau
 /tests/unit/http_/ @thrau
 
+# Runtime
+/localstack/runtime/ @thrau
+
+# Logging
+/localstack/logging/ @thrau @alexrashed @dominikschubert
```
Comment (thrau) [inline CODEOWNERS:71]: i'm happy to hand off code ownership of logging. @dfangl did a lot of the initial implementation

File: CODEOWNERS [hunk 3]
```diff
@@ -5,6 +5,26 @@
...
+/LICENSE.txt @HarshCasper @alexrashed
+
+# Docker
+/bin/docker-entrypoint.sh
+/.dockerignore
+/Dockerfile @alexrashed
+
+# Git, Pipelines, GitHub config
+/.circleci @alexrashed @dfangl @thrau
+/.github @alexrashed @dfangl @thrau
```
Comment (thrau) [inline CODEOWNERS:22]: maybe we should swap me out for @dominikschubert here?

**Review by thrau — COMMENTED** [review]
thanks for updating! added a few comments

File: CODEOWNERS (hunk 1 above)
Comment (alexrashed) [reply CODEOWNERS]: Whoops, missed the two, thanks!

**Comment by alexrashed** [comment]
Thanks @thrau for the review and the comments!
@dominikschubert and @dfangl: Could you please take a quick look and confirm that these changes are okay for you?
Thanks!

**Review by dfangl — APPROVED** [review]

**Review by dominikschubert — APPROVED** [review]

**Pull request merged**